Apps/Database/src/Delegates/DBProfileDelegate.cs

```diff
@@ -279,5 +278,23 @@ public DBResult<IEnumerable<UserProfileHistory>> GetUserProfileHistories(string
             result.Status = DBStatusCode.Read;
             return result;
         }
+
+        /// <inheritdoc/>
+        public IDictionary<string, int> GetLoggedInUserYearOfBirthCounts(DateTime startDate, DateTime endDate)
+        {
+            Dictionary<string, int> yobCount = this.dbContext.UserProfile
+                .Select(x => new { x.HdId, x.LastLoginDateTime, x.YearOfBirth })
+                .Concat(
+                    this.dbContext.UserProfileHistory.Select(x => new { x.HdId, x.LastLoginDateTime, x.YearOfBirth }))
+                .Where(x => x.YearOfBirth != null && x.LastLoginDateTime >= startDate && x.LastLoginDateTime <= endDate)
+                .Select(x => new { x.HdId, lastLoginDate = GatewayDbContext.DateTrunc("days", x.LastLoginDateTime), x.YearOfBirth })
+                .Distinct()
+                .GroupBy(x => x.YearOfBirth)
+                .Select(x => new { yearOfBirth = x.Key, count = x.Count() })
+                .OrderBy(x => x.yearOfBirth)
+                .ToDictionary(x => x.yearOfBirth!, x => x.count);
```

Comment:
The Select() and OrderBy() are not needed.

Reply:
I'm pretty sure that's how I get the key and value for the dictionary.  The orderby is to return the list in order of year of birth for a preliminary sort.

Reply:
ToDictionary() should be able to perform the same transformation that the Select() is doing, i.e.

```
ToDictionary(x => x.Key!, x => x.Count())
```

Using OrderBy() may alter the results, but it is not reliable since Dictionaries have no guaranteed order.  If we wanted the results to be sorted, we could return a SortedDictionary, passing in the unsorted Dictionary to its constructor.

Reply:
I removed the OrderBy.
I tried doing the ToDictionary() as you suggested but I get an error that it can't interpret x.YearOfBirth.  I think it needs the select as part of the query to get the result then it can be put in the dictionary.
I have asked Nino what he wants to do for sorting. If it is needed I will use a SortedDictionary.

Reply:
It looks like having the Select() statement is preventing a LINQ to SQL issue so keeping it seems fine.